### PR TITLE
fix: v3 Roi calculator display part2

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
The focus of this PR is to update the ROI calculation logic and improve the UI components in the RoiCalculator feature.

### Detailed summary
- Updated ROI calculation formulas for better accuracy
- Improved UI components in Details and RoiCalculator
- Added combined metrics like `combinedApy` and `combinedRate`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/widgets-internal/roi/aprHelper.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import { Percent } from "@pancakeswap/sdk";
-import { formatFraction } from "@pancakeswap/utils/formatFractions";
-
-export function getAccrued(principal: number, apr: Percent, timesCompounded = 1, stakeFor = 1) {
-  const aprAsDecimal = parseFloat(formatFraction(apr.asFraction, 6) || "0");
+export function getAccrued(principal: number, apr: number, compoundEvery = 0, stakeFor = 1, compoundOn = false) {
```
Comment (chefjackson) [inline packages/widgets-internal/roi/aprHelper.ts:1]: do we need the `compoundOn` parameter? Does it mean the same if we set `compoundEvery = 0`?

File: packages/widgets-internal/roi/aprHelper.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   const daysAsDecimalOfYear = stakeFor / 365;
-  return principal * (1 + aprAsDecimal / timesCompounded) ** (timesCompounded * daysAsDecimalOfYear);
+  const timesCompounded = 365 / compoundEvery;
```
Comment (chefjackson) [inline packages/widgets-internal/roi/aprHelper.ts:3]: what if we compound every 2 years?

Comment (chefjackson) [reply packages/widgets-internal/roi/aprHelper.ts:3]: for example stake for 4 years while compound every 2 years?

File: packages/widgets-internal/roi/aprHelper.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
...
+  if (timesCompounded !== 0 && compoundEvery <= stakeFor && compoundOn) {
+    return principal * (1 + apr / timesCompounded) ** (timesCompounded * daysAsDecimalOfYear);
+  }
+
+  return principal + principal * apr * daysAsDecimalOfYear; // simple calc when not compounding
```
Comment (ChefMomota) [inline packages/widgets-internal/roi/aprHelper.ts:8]: Avoid floating numbers, pls use bigNumber to handle it.

File: packages/widgets-internal/roi/aprHelper.ts (hunk 1 above)
Comment (Chef-Yogi) [reply packages/widgets-internal/roi/aprHelper.ts:1]: yes,let me change it

File: packages/widgets-internal/roi/aprHelper.ts (hunk 3 above)
Comment (Chef-Yogi) [reply packages/widgets-internal/roi/aprHelper.ts]: done

File: packages/widgets-internal/roi/aprHelper.ts (hunk 2 above)
Comment (Chef-Yogi) [reply packages/widgets-internal/roi/aprHelper.ts:5]: let me check with PM

Comment (Chef-Yogi) [reply packages/widgets-internal/roi/aprHelper.ts:3]: actually we don;t have that option on UI

**Review by ChefJerry — APPROVED** [review]

**Pull request merged**